darshan-util/pydarshan/tests/test_heatmap_handling.py

```diff
@@ -0,0 +1,561 @@
+                    [2099376],
+                ]
+            ),
+        ),
+        (
+            "examples/example-logs/ior_hdf5_example.darshan",
+            10,
+            ["read", "write"],
+            np.array(
+                [
+                    [
+                        0,
+                        827385.0734944909,
+                        50822.506110427385,
+                        50822.50611042739,
+                        50822.50611042739,
+                        50822.50611042737,
+                        17900.90206379957,
+                        30.830417529761142,
+                        1051289.1695824703,
+                        0,
+                    ],
+                    [
+                        0,
+                        307668.84624124144,
+                        50221.842911882275,
+                        50221.84291188228,
+                        50221.84291188228,
+                        50221.84291188226,
+                        540019.7821112294,
+                        418.49887595643366,
+                        108234.18114341467,
+                        942483.3199806289,
+                    ],
+                    [
+                        0,
+                        567426.7192208751,
+                        50754.88456134755,
+                        50754.88456134756,
+                        50754.88456134756,
+                        50754.884561347535,
+                        141846.72456250372,
+                        136490.77078712088,
+                        1050536.24718411,
+                        0,
+                    ],
+                    [
+                        0,
+                        1048576,
+                        0,
+                        0,
+                        0,
+                        0,
+                        0,
+                        251.0322619047617,
+                        1050548.9677380952,
+                        0,
+                    ],
+                ]
+            ),
+        ),
+    ],
+)
+def test_get_heatmap_data(
+    filepath,
+    expected_hmap_data,
+    xbins,
+    ops,
+):
+    # regression test for `heatmap_handling.get_heatmap_data()`
+
+    # generate the report and use it to obtain the aggregated data
+    report = darshan.DarshanReport(filepath)
+    agg_df = heatmap_handling.get_aggregate_data(
+        report=report, mods=["DXT_POSIX"], ops=ops
+    )
+    # run the aggregated data through the heatmap data code
+    actual_hmap_data = heatmap_handling.get_heatmap_data(agg_df=agg_df, xbins=xbins)
+
+    if filepath == "tests/input/sample-dxt-simple.darshan":
+        # check the data is conserved
+        assert actual_hmap_data.sum() == 4040
+        # make sure the output array is the correct shape
+        assert actual_hmap_data.shape == (1, xbins)
+        # make sure the output data contains identical values
+        assert_array_equal(actual_hmap_data, expected_hmap_data)
```

Comment:
here and anywhere else that does a comparison of NumPy arrays that may contain floating point values should use `assert_allclose` to avoid potential issues with floating point imprecision

Reply:
Yes, this was a good point and the newest version uses `assert_allclose()` for these cases. I'll resolve this.